tasks/rh-sign-cosign-image/rh-sign-cosign-image.yaml

```diff
@@ -0,0 +1,119 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: rh-sign-cosign-image
+  labels:
+    app.kubernetes.io/version: "1.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to sign container images in a Snapshot spec using the cosign.
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: "snapshot_spec.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+      script: |
+        #!/usr/bin/env bash
+        if [ "$TEST" != "1" ]; then
```

Comment:
If I understood it correctly mock script is inserted before the task script. Even if I append the mock script instead, it wouldn't help. And any more complicated modifications of the task script could cause more confusion.

Reply:
Correct, mock script is typically inserted at the top. That way if the mock defines a function called `cosign`, during test execution when the task is running, the mock function is called instead of real `cosign`. See the examples I linked.

> And any more complicated modifications of the task script could cause more confusion.

The point is that you don't need to modify the task script. Instead, the mock will override all the commands that we don't want to be called in the test (using function definitions). And then the test can check afterwards that those commands were called X number of times with all the expected parameters - just like when you use mocks in python tests.